pkg/agent/informant.go

```diff
@@ -746,6 +751,17 @@ func (s *InformantServer) handleSuspend(ctx context.Context, body *api.SuspendAg
 		return nil, 400, errors.New("Cannot suspend agent that is not yet registered")
 	}
 
+	// Acquire s.runner.requestLock so that when we return, we can guarantee that
+	locked = false
+	s.runner.lock.Unlock()
+
+	if err := s.runner.requestLock.TryLock(ctx); err != nil {
+		err = fmt.Errorf("Context expired while trying to acquire requestLock: %w", err)
+		s.runner.logger.Errorf("%s", err)
+		return nil, 500, err
```

Comment:
this is effectively a timeout (or maybe with a bit of extra formatting it could be?) is 500 the right code?

Reply:
> this is effectively a timeout

yep

> is 500 the right code?

I don't know. My motivating thought was that we have a lot of request timeouts for things that happen _while_ holding `requestLock`, so waiting for longer than expected probably means that something went wrong in the autoscaler-agent.

And if the request gets cancelled early and it's not the autoscaler-agent's fault, then the VM informant won't receive the 500 anyways.

Reply:
I think it's fine. This is more of a server-side timeout, so probably clients will get it. 

The thing, perhaps unnecessarily that I'm questioning is "if this is a 500 and we start keeping metrics on number of 500s" does this become noise?

Reply:
That's a fair concern. My thinking is that _generally_ we're expecting not to violate client-side timeouts, and if we do, that's primarily on the autoscaler-agent, not the vm-informant. So even if 500 isn't _technically_ the correct code from a metrics perspective, the _vibe_ of "which component messed up" is correct.

That said, my reasoning may be faulty.